File: Models/Photo.cs
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace MyAlbum.Models
{
    public class Photo
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public string FilePath { get; set; }

        public ICollection<Comment> Comments { get; set; }
        public ICollection<PhotoCategory> PhotoCategories { get; set; }

        public Album Album { get; set; }

        public User Author { get; set; }

        public Photo()
        {
            Comments = new Collection<Comment>();
            PhotoCategories = new Collection<PhotoCategory>();
        }
    }
}

File: MyAlbum.WebSPA/ClientApp/src/app/models/photo.ts
import { SaveAlbum } from './album';
import { Comment } from './comment';
import { User } from './user';

export interface PhotoCategory {
    id: number;
    name: string;
}

export interface Photo {
    id: number;
    name: string;
    filePath: string;
    boundingBoxFilePath: string;
    width: number;
    height: number;
    photoCategories: PhotoCategory[];
    locLat: number;
    locLng: number;
    centerLat: number;
    centerLng: number;
    mapZoom: number;
    comments: Comment[];
    totalComments: number;
    createdDate: Date;
    modifiedDate: Date;
    author: User;
    album: SaveAlbum;
}

export interface SavePhoto {
    id: number;
    name: string;
    locLat: number;
    locLng: number;
    centerLat: number;
    centerLng: number;
    mapZoom: number;
    photoCategories: PhotoCategory[];
    album: SaveAlbum;
}

export interface PositionModel {
    latitude: number;
    longitude: number;
}